hoodie-client/src/test/java/com/uber/hoodie/io/TestHoodieCleaner.java

```diff
@@ -17,31 +17,49 @@
 package com.uber.hoodie.io;
 
 import com.uber.hoodie.common.model.HoodieCleaningPolicy;
-import com.uber.hoodie.common.table.HoodieTableMetaClient;
-import com.uber.hoodie.common.table.timeline.HoodieInstant;
-import com.uber.hoodie.config.HoodieWriteConfig;
+import com.uber.hoodie.common.model.HoodieTableType;
 import com.uber.hoodie.common.model.HoodieTestUtils;
+import com.uber.hoodie.common.table.HoodieTableMetaClient;
 import com.uber.hoodie.common.util.FSUtils;
-
 import com.uber.hoodie.config.HoodieCompactionConfig;
+import com.uber.hoodie.config.HoodieWriteConfig;
 import com.uber.hoodie.table.HoodieTable;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
 import java.io.IOException;
-import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.UUID;
 
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
 
 /**
  * Tests around Cleaning logic in Hoodie
  */
+@RunWith(Parameterized.class)
 public class TestHoodieCleaner {
 
     private String basePath = null;
     private String[] partitionPaths = {"2016/01/01", "2016/02/02"};
     private HoodieTableMetaClient metaClient;
+    private HoodieTableType tableType;
+
+    @Parameterized.Parameters(name = "HoodieTableTypes")
```

Comment:
Does not make sense to parameterize this - becuase you anyways have a condition block for each of these options. Just create 2 different tests.